### DISCUSSION TITLE
Setting the inner text of a link node

### DISCUSSION THREAD

**Discussion by lorenzo-dallamuta** [ueberdosis/tiptap#5608] [General Questions]
I'm trying to build a form to insert a new Link, the behavior I intend to achieve is that I'll have an optional `Title` input in a form, then I'll check if the editor had a text selection when opening the form

- if a selection is present I will set the contents its to the title if a value is given, or to the current value of the selection, or to the value of an url field
- if a selection not is present I will set the contents its to the title if a value is given or to the value of an url field

my form submit handler looks something like this

```
const { from, to } = editor.state.selection;

editor
  .chain()
  .focus()
  .insertContentAt(
    { from, to },
      title ?? editor.state.doc.textBetween(from, to) ?? link,
    )
  .selectParentNode()
  .extendMarkRange("link")
  .setLink({ href: link })
  .run();
```

My issue is that the `selectParentNode` method is going to do what it's designed to do 😅, so that if I have a node that looks like `<p>this is a paragraph</p>` and select the text `"paragraph"` the whole paragraph node is being made into a Link mark, but if I skip the `selectParentNode` method nothing is marked as Link.

In other words: it seems that by invoking `insertContent` I'm losing the current selection (which makes sense since the length of the text has probably changed) how can I select what I just inserted?

**Comment by lorenzo-dallamuta** [comment]
I managed to produce the behaviour I wanted this way:

```
const { from, to } = editor.state.selection;

editor
  .chain()
  .focus()
  .extendMarkRange("link")
  .setLink({ href: link })
  .insertContentAt(
    { from, to },
      title ?? editor.state.doc.textBetween(from, to) ?? link,
  )
  .run();
```

The idea is to use the overload of `insertContent` that just replaces text and the text selection will still be available in the same positions after setting the link node.
